src/pages/networks/NetworkForwards.tsx

```diff
@@ -0,0 +1,165 @@
+import { fetchNetworkForwards } from "api/network-forwards";
+import { useDocs } from "context/useDocs";
+import DeleteNetworkForwardBtn from "pages/networks/actions/DeleteNetworkForwardBtn";
+import { Link } from "react-router-dom";
+import ExpandableList from "components/ExpandableList";
+import NetworkForwardPort from "pages/networks/NetworkForwardPort";
+
+interface Props {
+  network: LxdNetwork;
+  project: string;
+}
+
+const NetworkForwards: FC<Props> = ({ network, project }) => {
+  const docBaseLink = useDocs();
+  const notify = useNotify();
+
+  const {
+    data: forwards = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.networks, network, queryKeys.forwards, project],
+    queryFn: () => fetchNetworkForwards(network.name, project),
+  });
+
+  if (error) {
+    notify.failure("Loading network forwards failed", error);
+  }
+
+  const hasNetworkForwards = forwards.length > 0;
+
+  const headers = [
+    { content: "Listen address", sortKey: "listenAddress" },
+    { content: "Description", sortKey: "description" },
+    { content: "Default target address", sortKey: "defaultTarget" },
+    { content: "Ports", sortKey: "ports" },
+    { "aria-label": "Actions", className: "u-align--right actions" },
+  ];
+
+  const rows = forwards.map((forward) => {
+    return {
+      columns: [
+        {
+          content: forward.listen_address,
+          role: "cell",
+          "aria-label": "Listen address",
+        },
+        {
+          content: forward.description,
+          role: "cell",
+          "aria-label": "Description",
+        },
+        {
+          content: forward.config.target_address,
+          role: "cell",
+          "aria-label": "Default target address",
+        },
+        {
+          content: (
+            <ExpandableList
+              items={forward.ports.map((port) => (
+                <NetworkForwardPort key={port.listen_port} port={port} />
+              ))}
+            />
+          ),
+          role: "cell",
+          "aria-label": "Forwarded ports",
+        },
+        {
+          content: (
+            <>
+              <Link
+                className="p-button--base u-no-margin--bottom has-icon"
+                to={`/ui/project/${project}/networks/detail/${network.name}/forwards/${forward.listen_address}/edit`}
+              >
+                <Icon name="edit" />
+              </Link>
+              <DeleteNetworkForwardBtn
+                key={forward.listen_address}
+                network={network}
+                forward={forward}
+                project={project}
+              />
+            </>
+          ),
+          role: "rowheader",
+          className: "u-align--right actions",
+          "aria-label": "Actions",
+        },
+      ],
+      sortData: {
+        listenAddress: forward.listen_address,
+        description: forward.description,
+        defaultTarget: forward.config.target_address ?? "",
+        ports: forward.ports.length,
```

Comment:
The ports column displays all ports as text, but we are sorting the column by count. Is this not slightly confusing for the users? WDYT?

Reply:
True, and there is no right solution sorting for multivalued fields. I'll remove the sort by port option.